cdap-master/src/main/java/co/cask/cdap/data/runtime/main/ClientVersions.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.data.runtime.main;
+
+import kafka.Kafka;
+import org.apache.hadoop.util.VersionInfo;
+import org.apache.zookeeper.version.Info;
+
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Provides client versions of infrasturcture components.
+ */
+public class ClientVersions {
+
+  public static String getHadoopVersion() {
+    return VersionInfo.getVersion();
+  }
+
+  public static String getZooKeeperVersion() {
+    return String.format("%d.%d.%d.%d", Info.MAJOR, Info.MINOR, Info.MICRO, Info.REVISION);
+  }
+
+  public static String getKafkaVersion() {
+    // return AppInfoParser.getVersion();
+    URL kafkaJar = Kafka.class.getResource("/" + Kafka.class.getName().replace(".", "/") + ".class");
+    String[] tokens = kafkaJar.getPath().split("!")[0].split("/");
+    String jarFilename = tokens[tokens.length - 1];
+    Matcher matcher = Pattern.compile("kafka_2.10-(\\d+\\.\\d+\\.\\d+\\.\\d+)\\.jar").matcher(jarFilename);
```

Comment:
doesn't matter now since this is only called once, but in general its better to have the Pattern as a private static final of the class, since compiling a Pattern is relatively expensive.